src/microbe_stage/OrganelleDefinition.cs

```diff
@@ -657,6 +657,30 @@ private Vector3 CalculateCenterOffset()
         return offset;
     }
 
+    private LoadedSceneWithModelInfo? TryGetGraphicsForUpgrade(OrganelleUpgrades? upgrades)
+    {
+        if (upgrades != null)
+        {
+            foreach (var availableUpgrade in AvailableUpgrades)
+            {
+                var upgradeGraphics = availableUpgrade.Value.TryGetGraphicsScene();
+
+                if (upgradeGraphics != null)
+                {
+                    if (upgrades.UnlockedFeatures.Contains(availableUpgrade.Key))
+                    {
+                        if (availableUpgrade.Value.TryGetGraphicsScene().HasValue)
+                        {
+                            return availableUpgrade.Value.TryGetGraphicsScene();
```

Comment:
Here, refactoring the TryGet method in the upgrade to directly return the data like TryGetGraphicsScene would allow simplifying this and removing the duplicate method call (or actually considering line 666 there's actually 3 calls to the same method here).

Reply:
Now only the thing I commented on somewhere above and this (if it's necesary) yet to go.